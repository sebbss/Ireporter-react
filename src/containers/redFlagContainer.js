import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import RedFlag from '../components/redflags';
import redFlagAction from '../actions/redFlagActions';
import { Authenticated } from '../routes/authenticated';

export class RedFlagView extends Component {
    state = {
      payload: [],
    }

    componentWillMount() {
      const { flags } = this.props;
      flags();
    }

    componentWillReceiveProps(nextProps) {
      const { redFlags: { payload } } = nextProps;
      this.setState({
        payload,
      });
    }

    render() {
      const { payload } = this.state;
      return (
        <div className="row container-fluid p-2">
          { payload ? payload.map(redFlag => (
            <RedFlag
              flagType={redFlag.flag_type}
              description={redFlag.description}
            />
          )) : <p className="payload">No Red-flags </p>}
        </div>

      );
    }
}
export const mapStateToProps = state => ({
  redFlags: state.redFlagReducer,
});

RedFlagView.propTypes = {
  flags: PropTypes.func.isRequired,
  redFlags: PropTypes.shape().isRequired,
};

export default connect(mapStateToProps, { flags: redFlagAction })(Authenticated(RedFlagView));
