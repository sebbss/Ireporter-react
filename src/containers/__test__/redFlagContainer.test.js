import React from 'react';
import { shallow } from 'enzyme';
import { RedFlagView } from '../redFlagContainer';


const props = {
  flags: jest.fn(),
  redFlags: { payload: [] },
};
const nextProps = {
    redFlags: {
        payload: [{created_by:2}],
  },
};

describe('socialAuthView', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<RedFlagView {...props} />);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should change state', () => {
    wrapper.setProps({ ...nextProps });
    expect(wrapper.instance().state.payload).toEqual([{created_by:2}]);
  });
});
