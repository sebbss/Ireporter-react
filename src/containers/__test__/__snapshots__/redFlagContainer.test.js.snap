// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`socialAuthView should match snapshot 1`] = `ShallowWrapper {}`;
